clippy_lints/src/bool_assert_comparison.rs

```diff
@@ -1,3 +1,4 @@
+#![allow(clippy::compressable_if)]
```

Comment:
I'm a bit wary to add a complexity lint that needs multiple `allow` annotations *in our own code*.

I'd either see the lint applied everywhere we can (which would show if it actually is an improvement), or keep it in nursery or pedantic for now.

The last thing we need is a lint that will trigger a lot on average code with a warning that has no urgency at all.